Sources/AWSLambdaRuntime/LambdaRuntimeClient.swift

```diff
@@ -596,18 +582,29 @@ private final class LambdaChannelHandler<Delegate: LambdaChannelHandlerDelegate>
     func writeResponseBodyPart(
         isolation: isolated (any Actor)? = #isolation,
         _ byteBuffer: ByteBuffer,
-        requestID: String
+        requestID: String,
+        hasCustomHeaders: Bool
     ) async throws {
         switch self.state {
         case .connected(_, .waitingForNextInvocation):
             fatalError("Invalid state: \(self.state)")
 
         case .connected(let context, .waitingForResponse):
             self.state = .connected(context, .sendingResponse)
-            try await self.sendResponseBodyPart(byteBuffer, sendHeadWithRequestID: requestID, context: context)
+            try await self.sendResponseBodyPart(
+                byteBuffer,
+                sendHeadWithRequestID: requestID,
+                context: context,
+                hasCustomHeaders: hasCustomHeaders
+            )
 
         case .connected(let context, .sendingResponse):
-            try await self.sendResponseBodyPart(byteBuffer, sendHeadWithRequestID: nil, context: context)
+            try await self.sendResponseBodyPart(
```

Comment:
It is invalid to be have custom headers here, perhaps you should add a precondition for that, or throw an error